contracts/tokenbridge/test/TestCustomTokenL1.sol

```diff
@@ -103,6 +131,63 @@ contract TestCustomTokenL1 is aeERC20, ICustomToken {
     }
 }
 
+contract TestOrbitCustomTokenL1 is TestCustomTokenL1 {
+    using SafeERC20 for IERC20;
+
+    constructor(address _gateway, address _router) TestCustomTokenL1(_gateway, _router) {}
+
+    function registerTokenOnL2(
+        address l2CustomTokenAddress,
+        uint256 maxSubmissionCostForCustomGateway,
+        uint256 maxSubmissionCostForRouter,
+        uint256 maxGasForCustomGateway,
+        uint256 maxGasForRouter,
+        uint256 gasPriceBid,
+        uint256 valueForGateway,
+        uint256 valueForRouter,
+        address creditBackAddress
+    ) public payable override {
+        // we temporarily set `shouldRegisterGateway` to true for the callback in registerTokenToL2 to succeed
+        bool prev = shouldRegisterGateway;
+        shouldRegisterGateway = true;
+
+        // transfer fees from user to here, and approve router to use it
+        {
+            address inbox = IOrbitGatewayRouter(router).inbox();
+            address bridge = address(IInbox(inbox).bridge());
+            address nativeToken = IERC20Bridge(bridge).nativeToken();
+
+            IERC20(nativeToken).safeTransferFrom(
+                msg.sender,
+                address(this),
+                valueForGateway + valueForRouter
+            );
+            IERC20(nativeToken).approve(router, valueForRouter);
```

Comment:
All this value gets used right? If it doesnt we should probably set approval to 0 at the end of this func

Reply:
Good point, value is not totally used in case Inbox would for some reason already hold some amount of native tokens. Added re-setting to 0